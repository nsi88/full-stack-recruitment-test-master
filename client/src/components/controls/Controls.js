import React from 'react';
import './Controls.scss';
import priceAlerts from '../../icons/price-alerts.svg';

const Controls = () => (
  <div className='controls'>
    <div>
      <span className='control'>Filter</span>
      <span className='control'>Sort</span>
    </div>
    <div>
      <span className='control'>
        <img className='priceAlerts' role='presentation' src={priceAlerts}/>
        Price alerts
      </span>
    </div>
  </div>
);

export default Controls;
