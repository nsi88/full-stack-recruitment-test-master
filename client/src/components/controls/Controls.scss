@import "~bpk-mixins";

.controls {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: $bpk-spacing-sm;
  background-color: $bpk-color-white;
}

.control {
  @include bpk-text-base();
  color: $bpk-color-blue-500;
  cursor: pointer;
}

.control:not(:first-child) {
  margin-left: 2.375rem;
}

.price-alerts {
  width: $bpk-icon-size-sm;
  height: $bpk-icon-size-sm;
  position: relative;
  top: .175rem;
  left: -0.175rem;
}
