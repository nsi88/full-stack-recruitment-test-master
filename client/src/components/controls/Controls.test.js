import React from 'react';
import Controls from '../Controls';
import renderer from 'react-test-renderer';

test('render', () => {
  const tree = renderer.create(<Controls/>).toJSON();
  expect(tree).toMatchSnapshot();
});
