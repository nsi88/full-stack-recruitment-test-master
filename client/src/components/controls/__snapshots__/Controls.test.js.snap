// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="controls"
>
  <div>
    <span
      className="controls__control"
    >
      Filter
    </span>
    <span
      className="controls__control"
    >
      Sort
    </span>
  </div>
  <div>
    <span
      className="controls__control control"
    >
      <img
        className="control__price-alerts"
        role="presentation"
        src="test-file-stub"
      />
      Price alerts
    </span>
  </div>
</div>
`;
